base/auspice_export.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import division, print_function
+import sys, os, time, gzip, glob
+from base.io_util import make_dir, remove_dir, tree_to_json, write_json, myopen
+import json
+from pdb import set_trace
+from collections import defaultdict
+
+def export_frequency_json(self, prefix, indent):
+
+    # local function or round frequency estimates to useful precision (reduces file size)
+    def process_freqs(freq):
+        return [round(x,4) for x in freq]
+
+    # construct a json file containing all frequency estimate
+    # the format is region_protein:159F for mutations and region_clade:123 for clades
+    if hasattr(self, 'pivots'):
+        freq_json = {'pivots':process_freqs(self.pivots)}
+        if hasattr(self, 'mutation_frequencies'):
+            freq_json['counts'] = {x:list(counts) for x, counts in self.mutation_frequency_counts.iteritems()}
+            for (region, gene), tmp_freqs in self.mutation_frequencies.iteritems():
+                for mut, freq in tmp_freqs.iteritems():
+                    label_str =  region+"_"+ gene + ':' + str(mut[0]+1)+mut[1]
+                    freq_json[label_str] = process_freqs(freq)
+        # repeat for clade frequencies in trees
+        if hasattr(self, 'tree_frequencies'):
+            for region in self.tree_frequencies:
+                for clade, freq in self.tree_frequencies[region].iteritems():
+                    label_str = region+'_clade:'+str(clade)
+                    freq_json[label_str] = process_freqs(freq)
+        # repeat for named clades
+        if hasattr(self, 'clades_to_nodes') and hasattr(self, 'tree_frequencies'):
+            for region in self.tree_frequencies:
+                for clade, node in self.clades_to_nodes.iteritems():
+                    label_str = region+'_'+str(clade)
+                    freq_json[label_str] = process_freqs(self.tree_frequencies[region][node.clade])
+        # write to one frequency json
+        if hasattr(self, 'tree_frequencies') or hasattr(self, 'mutation_frequencies'):
+            write_json(freq_json, prefix+'_frequencies.json', indent=indent)
+    else:
+        self.log.notify("Cannot export frequencies - pivots do not exist")
+
+
+def summise_publications_from_tree(tree):
```

Comment:
Should this be named `summarise_publications_from_tree ` instead? Or `summarize_publications_from_tree `, if you want to Americanize it.